spec/features/user_adds_book_review_spec.rb

```diff
@@ -0,0 +1,42 @@
+require 'rails_helper'
+
+feature 'User adds book review' do
+  scenario 'successfully' do
+    user = create(:user)
+    book = create(:book)
+
+    visit book_path(book, as: user)
+    click_link 'Add your review'
+    fill_in 'review_body', with: 'this book was amazing'
+    click_button 'Save'
+
+    expect(page).to have_content 'Thanks for adding your review'
+  end
+
+  scenario 'without authentication' do
+    book = create(:book)
+
+    visit book_path(book)
+
+    expect(page).to_not have_review_link
```

Comment:
This should be a good candidate to test it in lower level like either view spec or a method under presenter class.